sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -260,16 +260,16 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
         Nil
       }
 
-    case DescribeTable(r: DataSourceV2Relation, isExtended) =>
-      DescribeTableExec(r.table, isExtended) :: Nil
+    case r @ DescribeTable(datasource: DataSourceV2Relation, isExtended) =>
+      DescribeTableExec(r.output, datasource.table, isExtended) :: Nil
 
     case DropTable(catalog, ident, ifExists) =>
       DropTableExec(catalog, ident, ifExists) :: Nil
 
     case AlterTable(catalog, ident, _, changes) =>
       AlterTableExec(catalog, ident, changes) :: Nil
 
-    case r : ShowTables =>
+    case r: ShowTables =>
```

Comment:
Nit: this is a non-functional change that can cause commit conflicts. Please remove it.

Reply:
OK. Reverted this.